manager_42.py

```diff
@@ -32,55 +32,47 @@
 import sys
 import time
 import yaml
+from collections import namedtuple
 
 from osclib.memoize import memoize
 
 logger = logging.getLogger()
 
-OPENSUSE = 'openSUSE:Leap:42.3'
-
 makeurl = osc.core.makeurl
 http_GET = osc.core.http_GET
 http_DELETE = osc.core.http_DELETE
 http_PUT = osc.core.http_PUT
 http_POST = osc.core.http_POST
 
 class Manager42(object):
-    def __init__(self, from_prj, caching = True):
-        self.from_prj = from_prj
+
+    config_defaults = {
+        'project_preference_order' : [],
+        'from_prj' : 'openSUSE:Leap:42.3',
+        'factory' : 'openSUSE:Factory',
+        }
+
+    def __init__(self, caching = True, configfh = None):
         self.caching = caching
         self.apiurl = osc.conf.config['apiurl']
-        self.project_preference_order = [
-                'SUSE:SLE-12-SP3:GA',
-                'SUSE:SLE-12-SP2:Update',
-                'SUSE:SLE-12-SP2:GA',
-                'SUSE:SLE-12-SP1:Update',
-                'SUSE:SLE-12-SP1:GA',
-                'SUSE:SLE-12:Update',
-                'SUSE:SLE-12:GA',
-                'openSUSE:Leap:42.2:Update',
-                'openSUSE:Leap:42.2',
-                'openSUSE:Leap:42.2:NonFree:Update',
-                'openSUSE:Leap:42.2:NonFree',
-                'openSUSE:Leap:42.1:Update',
-                'openSUSE:Leap:42.1',
-                'openSUSE:Leap:42.1:NonFree:Update',
-                'openSUSE:Leap:42.1:NonFree',
-                'openSUSE:Factory',
-                'openSUSE:Factory:NonFree',
-                'openSUSE:Leap:42.3:SLE-workarounds',
-                'openSUSE:Leap:42.2:SLE-workarounds',
-                ]
-
-        self.parse_lookup(self.from_prj)
+        self.config = self._load_config(configfh)
+
+        self.parse_lookup(self.config.from_prj)
         self.fill_package_meta()
         self.packages = dict()
-        for project in [self.from_prj] + self.project_preference_order:
+        for project in [self.config.from_prj] + self.config.project_preference_order:
             self.packages[project] = self.get_source_packages(project)
 
+    # FIXME: add to ToolBase and rebase Manager42 on that
+    def _load_config(self, handle = None):
+        d = self.__class__.config_defaults
+        y = yaml.safe_load(handle) if handle is not None else {}
```

Comment:
The config is still read every-time the class is loaded, but presumably unlike `ReviewBot` this class is not instantiated multiple times so it may not matter. Although if they share the same code they might as well both check that not already loaded.